### PR TITLE
glTF skinning

### PR DESCRIPTION
**Pull request by AnastaZIuk**

### CODE REVIEW

**Comment by AnastaZIuk** [comment]
I'm switching to BaW, leaving this PR for you @Devsh-Graphics-Programming/seniors

File: include/nbl/builtin/cache/ICacheKeyCreator.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+				{
+					std::stringstream stream;
+					stream << std::setfill('0') << std::hex << value;
+					return stream.str();
+				};
+		};
+	}
+}
+
+#endif // __I_CACHE_KEY_CREATOR_H_INCLUDED__
```
[the remark this answers is not in the document]
Comment (devshgraphicsprogramming) [reply include/nbl/builtin/cache/ICacheKeyCreator.h:26]: @AnastaZIuk can you fix this?

File: include/nbl/builtin/glsl/transform_tree/debug/debug_draw.frag [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#version 440 core
+
+layout(location = 0) in vec4 inColor;
+layout(location = 0) out vec4 outColor;
+
+void main()
+{
+	outColor = inColor;
+}
```
[the remark this answers is not in the document]
Comment (devshgraphicsprogramming) [reply include/nbl/builtin/glsl/transform_tree/debug/debug_draw.frag:9]: should go into `nbl/builtin/glsl/material/debug`

check if it doesn't duplicate

File: include/nbl/builtin/cache/ICacheKeyCreator.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (AnastaZIuk) [reply include/nbl/builtin/cache/ICacheKeyCreator.h:26]: ofc,todo

File: include/nbl/builtin/glsl/transform_tree/debug/debug_draw.frag (hunk 2 above)
[the remark this answers is not in the document]
Comment (AnastaZIuk) [reply include/nbl/builtin/glsl/transform_tree/debug/debug_draw.frag:9]: todo

**Pull request merged**